conans/test/integration/test_pkg_signing.py

```diff
@@ -0,0 +1,47 @@
+import os
+import textwrap
+
+from conans.test.assets.genconanfile import GenConanfile
+from conans.test.utils.tools import TestClient
+from conans.util.files import save
+
+
+def test_pkg_sign():
+    c = TestClient(default_server_user=True)
+    c.save({"conanfile.py": GenConanfile("pkg", "0.1").with_exports("export/*")
+            .with_exports_sources("export_sources/*").with_package_file("myfile", "mycontents!"),
+            "export/file1.txt": "file1!",
+            "export_sources/file2.txt": "file2!"})
+    signer = textwrap.dedent(r"""
+        import os
+
+        def sign(ref, artifacts_folder, signature_folder):
+            print("Signing ref: ", ref)
+            print("Signing folder: ", artifacts_folder)
+            files = []
+            for f in sorted(os.listdir(artifacts_folder)):
```

Comment:
Might be possible that the `signature_folder` contains something that is signed again?

Reply:
Not sure what you mean, the code is getting just the files with:
```python
for f in sorted(os.listdir(artifacts_folder)):
                if os.path.isfile(os.path.join(artifacts_folder, f)):
                    files.append(f)
```
It is true that users could recurse into the ``download/metadata`` folder, and that is unexpected, that folder should explicitly be left out of signing, by definition.

Reply:
> It is true that users could recurse into the download/metadata folder, and that is unexpected, that folder should explicitly be left out of signing, by definition.

That is what I meant.

If that is that `by definiton` and you trust the users, then ok.

Reply:
Of course, I'd prefer this not be this way, but that would mean having the ``metadata`` folder in a different location, for the upload it might be possible, but still in the download they will find the ``metadata`` subfolder there, because while downloading from the server everything will be in this folder. So maybe it is better that users are aware of this fact from the very beginning, in both upload and download flows?

Reply:
ok